h2o-core/src/test/java/water/rapids/RapidsTest.java

```diff
@@ -28,18 +34,118 @@
 
 
 public class RapidsTest extends TestUtil {
-  @BeforeClass public static void setup() { stall_till_cloudsize(1); }
-
   @Rule
-  public transient ExpectedException ee = ExpectedException.none(); 
+  public transient ExpectedException ee = ExpectedException.none();
+
+  @BeforeClass
+  public static void setup() {
+    stall_till_cloudsize(1);
+  }
+
+  @Test
+  public void testSpearman() {
+    Scope.enter();
+    try {
+      final Frame frame = new TestFrameBuilder()
+              .withName("heightsweights")
+              .withVecTypes(Vec.T_NUM, Vec.T_NUM)
+              .withColNames("HEIGHT", "WEIGHT")
+              .withDataForCol(0, ard(175, 166, 170, 169, 188, 175, 176, 171, 173, 175, 173, 174, 169))
+              .withDataForCol(1, ard(69, 55, 67, 52, 90, 53, 57, 57, 68, 73, 62, 90, 63))
+              .build();
+
+      Val pearson = Rapids.exec("(spearman heightsweights 'HEIGHT' 'WEIGHT')");
+      assertTrue(pearson instanceof ValNum);
+
+      assertEquals(0.454357724505124, pearson.getNum(), 1e-8);
+    } finally {
+      Scope.exit();
+    }
+  }
+
+  @Test
+  public void testSpearmanIris() {
+    Scope.enter();
+    try {
+      final Frame iris = TestUtil.parse_test_file(Key.make("iris_spearman"), "smalldata/junit/iris.csv");
+      Scope.track_generic(iris);
+
+      final Val pearson_sepal = Rapids.exec("(spearman iris_spearman 'sepal_len' 'sepal_wid')");
+      assertTrue(pearson_sepal instanceof ValNum);
+      assertEquals(-0.1720027734934786, pearson_sepal.getNum(), 1e-8);
+
+    } finally {
+      Scope.exit();
+    }
+  }
   
-  @Test public void bigSlice() {
+  @Test
+  public void testSort(){
```

Comment:
This test demonstrates why the existing sort is not usable for us - it doesn't keep the original order and was probably made for different purpose by @wendycwong.

We need the original dataframe (or a copy of it), but with the original order of rows, only labeled. This is mandatory for Spearman.

Reply:
We need two vectors from the original frame ranked, without affecting the order of rows.